tools/scripts/rebuildSoundLibraryManifest.rb

```diff
@@ -0,0 +1,382 @@
+        metadata['aliases'] = aliases_by_key[key]
+        [key, metadata]
+      end
+    end
+  rescue
+    warn "There was a problem reading the existing manifest.  Rebuilding from scratch..."
+    return {}
+  end
+
+  # Given a map of S3 objects for sounds, build up an sound
+  # metadata map.
+  def build_sound_metadata(sound_objects, previous_metadata)
+    metadata_progress_bar = ProgressBar.create(total: sound_objects.size) unless @options[:verbose] || @options[:quiet]
+    sound_metadata_by_name = {}
+
+    # Parallelize metadata construction because some objects will require an
+    # extra S3 request to get version IDs or image dimensions.
+    Parallel.map(sound_objects.keys, finish: lambda do |name, _, result|
+      # This lambda runs synchronously after each entry is done processing - it's
+      # used to collect up results and warnings to the original process/thread.
+      if result.is_a? Hash
+        sound_metadata_by_name[name] = result
+      else
+        @warnings.push result
+      end
+      metadata_progress_bar.increment unless metadata_progress_bar.nil?
+    end
+) do |name|
+      # This is the parallel block.  This block should return a string to
+      # generate a warning and skip the sound, and a metadata Hash in
+      # the success case.
+      objects = sound_objects[name]
+
+      # Drop this sound if it is missing its JSON or MP3 components
+      if objects['json'].nil?
+        next "sound #{name} does not have a JSON file and was skipped."
+      elsif objects['mp3'].nil?
+        next "sound #{name} does not have a MP3 file and was skipped."
+      end
+
+      # Before we do anything else, check the last modify times on both files.
+      # If they are unchanged, just use the old metadata (+aliases).
+      # This saves us from actually downloading the files from S3
+      if previous_metadata[name].is_a?(Hash) &&
+          previous_metadata[name]['jsonLastModified'] == objects['json'].last_modified.to_s &&
+          previous_metadata[name]['mp3LastModified'] == objects['mp3'].last_modified.to_s
+        verbose "#{bold name} is unchanged, using old metadata"
+        next previous_metadata[name]
+      end
+
+      # Actually download the JSON from S3
+      begin
+        json_response = objects['json'].get
+        metadata = JSON.parse(json_response.body.read)
+      rescue Aws::Errors::ServiceError => service_error
+        next <<-WARN
+There was an error retrieving #{name}.json from S3:
+#{service_error}
+The sound has been skipped.
+        WARN
+      rescue JSON::JSONError => json_error
+        next <<-WARN
+There was an error parsing #{name}.json:
+#{json_error}
+The sound has been skipped.
+        WARN
+      end
+
+      # Verify that the parsed metadata contains all expected fields
+      next "sound #{name} is missing the 'name' attribute." unless metadata['name'].is_a?(String)
+      next "sound #{name} is missing the 'sound' attribute." unless metadata['time'].is_a?(Integer)
```

Comment:
I think this warning should say it's missing the 'time' attribute, not the 'sound' attribute.